frontend/src/lib/components/BranchButton.svelte

```diff
@@ -0,0 +1,384 @@
+	import { apiAddress } from '$lib/net';
+	import { ToastType, addToast } from '$lib/toast';
+	import { cache } from '$lib/cache';
+
+	let showMenu = false;
+	let protectedBranches: string[] = [];
+	let nonProtectedBranches: string[] = [];
+	let newBranchName: string = '';
+	let showInput = false;
+
+	const currentBranch = derived(branchName, ($branchName) => $branchName);
+
+	/**
+	 * Fetches existing branches from the GitHub API.
+	 *
+	 * This asynchronous function sends a GET request to retrieve the list of branches
+	 * from the specified GitHub repository. If the request is successful, it returns
+	 * a promise that resolves to an array containing the branch data. If the request
+	 * fails (e.g., due to network issues, authentication problems, or server errors),
+	 * it returns a promise that rejects with an error object containing details about the failure.
+	 *
+	 * @returns {Promise<Array>} A promise that resolves to an array of branches on success,
+	 * or rejects with an error object containing details about the failure.
+	 *
+	 * @throws Will throw an error if the response from the API is not successful. The error object
+	 * will include the status code and any additional error message from the API.
+	 *
+	 * @example
+	 * async function main() {
+	 *   try {
+	 *     const branches = await fetchExistingBranches();
+	 *     console.log('Existing branches:', branches);
+	 *   } catch (err) {
+	 *     console.error('Failed to fetch branches:', err);
+	 *   }
+	 * }
+	 */
+	async function fetchExistingBranches() {
+		const response = await fetch(`${apiAddress}/api/branches`, {
+			method: 'GET',
+			credentials: 'include',
+			headers: {
+				'Content-Type': 'application/json'
+			}
+		});
+
+		// Check if the response is successful
+		if (!response.ok) {
+			// Handle error without try-catch
+			response.json().then((errorMessage) => {
+				console.error('Failed to fetch branches:', errorMessage);
+				addToast({
+					message: `Error fetching branches: ${response.statusText}. ${JSON.stringify(errorMessage)}`,
+					type: ToastType.Error,
+					dismissible: true
+				});
+			});
+			return; // Exit if response is not OK
+		}
+
+		// Extract and set the branches if the response is successful
+		response.json().then((data) => {
+			if (data.data && data.data.branches) {
+				// Map through branches to filter out protected ones and extract branch names
+				nonProtectedBranches = data.data.branches
+					.filter((branch: string) => !branch.includes('(protected)')) // Filter out protected branches
+					.map((branch: string) => branch.split(' (')[0]); // Extract just branch names
+				protectedBranches = data.data.branches
+					.filter((branch: string) => branch.includes('(protected)')) // Filter out protected branches
+					.map((branch: string) => branch.split(' (')[0]); // Extract just branch names
+			} else {
+				protectedBranches = []; // Reset if no branches found
+				nonProtectedBranches = [];
+			}
+		});
```

Comment:
You could actually use <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch> the `.catch` method, my issue with `try` blocks is because you don't know exactly what's failing, or how it'll fail, whereas `catch` on a fetch method is pretty much always going to be "http request failed`. I don't know whether a fetch error is an actual "error" error, so who knows